File: gateway/gateway.py
import os

from flask import Flask, jsonify, request
import requests

app = Flask(__name__)

# Get api key from environment variable
OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")


@app.route('/v1/chat/completions', methods=['POST'])
def get_chat_completions():
    try:
        data = request.get_json()
        headers = {
            'Content-Type': 'application/json',
            'Authorization': f'Bearer {OPENAI_API_KEY}'
        }
        response = requests.post('https://api.openai.com/v1/chat/completions', json=data, headers=headers)
        return jsonify(response.json()), response.status_code
    except Exception as e:
        print(e)
        return 'Something went wrong', 500


if __name__ == '__main__':
    app.run(debug=True, port=3000)

File: frontend/gateway/gateway.js
const express = require('express');
const axios = require('axios');
const app = express();
const port = process.env.PORT || 3000;

app.use(express.json());

const OPENAI_API_KEY = process.env.OPENAI_API_KEY;

app.post('/v1/chat/completions', async (req, res) => {
    try {
        const {
            model,
            messages,
            temperature,
            top_p,
            n,
            stream,
            stop,
            max_tokens,
            presence_penalty,
            frequency_penalty,
            logit_bias,
            user
        } = req.body;

        const response = await axios.post('https://api.openai.com/v1/chat/completions', {
            model,
            messages,
            temperature,
            top_p,
            n,
            stream,
            stop,
            max_tokens,
            presence_penalty,
            frequency_penalty,
            logit_bias,
            user
        }, {
            headers: {
                'Content-Type': 'application/json',
                'Transfer-Encoding': 'chunked',
                'Authorization': `Bearer ${OPENAI_API_KEY}` // your API key
            }
        });

        res.send(response.data);
    } catch (error) {
        console.error(error);
        res.status(500).send('Something went wrong');
    }
});

app.listen(port, () => {
    console.log(`Server listening on port ${port}`);
});
